learning/tour-of-beam/learning-content/windowing/windowing-concept/description.md

```diff
@@ -0,0 +1,57 @@
+
+One of the main use cases for sliding time windows is to compute running aggregates. For example, if you want to compute a running average of the past 60 seconds’ worth of data updated every 30 seconds, you can use sliding time windows. This is done by defining a window duration of 60 seconds and a sliding interval of 30 seconds. With this configuration, you will have windows that slide every 30 seconds, each one covering a 60-second interval.
+
+Another use case for sliding time windows is to perform anomaly detection. By computing the running aggregates over a sliding window, you can detect patterns that deviate significantly from the historical data.
+
+Sliding time windows also allows to look at data in a more dynamic way. This is useful when you have a high-frequency data stream and you want to look at the most recent data.
+
+In summary, Sliding time windows are useful for performing running aggregations, anomaly detection and looking at data in a more dynamic way.
+
+
+`Session windows` are a type of windowing that groups data elements based on periods of inactivity or "gaps" in the data stream. They are useful when you want to group data elements that are related to a specific event or activity together.
+
+One of the main use cases for session windows is to group together data elements that are related to a user's session on a website or application. By using session windows with a relatively short gap duration, you can ensure that all the events related to a user's session are grouped together. This allows you to compute session-level metrics, such as the number of pages viewed per session, the duration of a session, or the number of events per session.
+
+Another use case for session windows is to group together data elements that are related to a specific device's usage. For example, if you are collecting sensor data, you can use session windows to group together data elements that are collected while the device is in use. This allows you to compute device-level metrics, such as the number of sensor readings per device, the duration of device usage, or the number of events per device.
+
+In summary, session windows are useful for grouping data elements that are related to specific events or activities, such as user sessions or device usage. This allows you to compute event- or device-level metrics.
+
+
+A `single global window` is a type of windowing that treats all data elements as belonging to the same window. This means that all elements in the data stream are processed together and no windowing is applied.
+
+The main use case for a single global window is when you want to process all the data elements in your data stream as a whole, without breaking them up into smaller windows. This can be useful in situations where you don't need to compute window-level metrics, such as running averages or counts, but instead want to process the entire data stream as a single unit.
```

Comment:
The primary use case for a single global window is when you want to process all the data elements in your data stream **as a whole** without breaking them up into smaller windows. For example, this can be useful when you don't need to compute window-level metrics, such as running averages or counts, but instead, you want to process the entire data stream as a single unit.